### ISSUE TITLE
Android splash screen image jumps slightly when loading

### ISSUE THREAD

**Issue by shwanton** [zoontek/react-native-bootsplash#20]
### 💻 My environment

```bash
    OS: macOS 10.14.6
    CPU: (4) x64 Intel(R) Core(TM) i7-6567U CPU @ 3.30GHz
    Memory: 54.65 MB / 16.00 GB
    Shell: 5.7.1 - /usr/local/bin/zsh
  Binaries:
    Node: 10.15.3 - /var/folders/fc/dg0hjk2n5lg7g8brjxyfhx6h0000gn/T/yarn--1567232771367-0.714416577935092/node
    Yarn: 1.17.3 - /var/folders/fc/dg0hjk2n5lg7g8brjxyfhx6h0000gn/T/yarn--1567232771367-0.714416577935092/yarn
    npm: 6.4.1 - ~/.asdf/installs/nodejs/10.15.3/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  SDKs:
    iOS SDK:
      Platforms: iOS 12.4, macOS 10.14, tvOS 12.4, watchOS 5.3
    Android SDK:
      API Levels: 23, 24, 27, 28
      Build Tools: 27.0.3, 28.0.3
      System Images: android-23 | Google APIs Intel x86 Atom, android-24 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.4 AI-183.6156.11.34.5692245
    Xcode: 10.3/10G8 - /usr/bin/xcodebuild
  npmPackages:
    @react-native-community/cli: ^2.9.0 => 2.9.0
    react: 16.9.0 => 16.9.0
    react-native: 0.61.0-rc.0 => 0.61.0-rc.0
```
### 🕵️‍♂️ Reproducing the issue

Splash screen on some Android devices [jumps slightly when loaded](https://github.com/crazycodeboy/react-native-splash-screen/issues/241)

### 🤞Solution

There are a couple solutions in this [issue](https://github.com/crazycodeboy/react-native-splash-screen/issues/241) but none of them seem to be working on all devices.

Wondering if anyone else is seeing the same behavior and has any good fixes.

**Comment by zoontek** [comment]
@shwanton Are you using this package or react-native-splash-screen?

**Comment by shwanton** [comment]
My bad, I didn't add a repro. I am using react-native-bootsplash, but I cross referenced the issue from react-native-splash-screen since I was seeing that issue here. I'll try taking a video to demonstrate what I'm seeing.

It could be that my `@mipmap/splash_logo` centering is not calculating the status bar since that seems to be the amount that it's jumping.

`AndroidManifest.xml`
```xml
  <application
      android:name=".MainApplication"
      android:label="@string/app_name"
      android:icon="@mipmap/ic_launcher"
      android:roundIcon="@mipmap/ic_launcher"
      android:allowBackup="false"
      android:theme="@style/AppTheme">

    <activity
        android:name=".MainActivity"
        android:label="@string/app_name"
        android:configChanges="keyboard|keyboardHidden|orientation|screenSize"
        android:windowSoftInputMode="adjustPan"
        android:exported="true" />

    <activity
        android:name="com.zoontek.rnbootsplash.RNBootSplashActivity"
        android:theme="@style/BootTheme">
      <intent-filter>
        <action android:name="android.intent.action.MAIN" />
        <category android:name="android.intent.category.LAUNCHER" />
      </intent-filter>
    </activity>

    <activity android:name="com.facebook.react.devsupport.DevSettingsActivity" />

  </application>
```

`styles.xml`
```xml
<resources>
  <!-- Base application theme. -->
  <style
      name="AppTheme"
      parent="Theme.AppCompat.Light.NoActionBar">
    <!-- Customize your theme here. -->
    <item name="android:textColor">#000000</item>
  </style>

  <style
      name="BootTheme"
      parent="AppTheme">
    <!-- set bootsplash.xml as activity background -->
    <item name="android:windowBackground">@drawable/bootsplash</item>
  </style>
</resources>

```

`@drawable/bootsplash.xml`
```xml
<layer-list
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:opacity="opaque">
  <item android:drawable="@drawable/splash_gradient" />
  <item android:gravity="center">
    <bitmap android:src="@mipmap/splash_logo" />
  </item>
</layer-list>

```

`@drawable/splash_gradient`
```xml
<layer-list
    xmlns:android="http://schemas.android.com/apk/res/android"
    android:opacity="opaque">
  <item android:drawable="@drawable/splash_gradient" />
  <item android:gravity="center">
    <bitmap android:src="@mipmap/splash_logo" />
  </item>
</layer-list>
```

`MainActivity.java`
```java
...
import com.zoontek.rnbootsplash.RNBootSplash;

public class MainActivity extends ReactActivity {
    ....
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        RNBootSplash.show(R.drawable.bootsplash, MainActivity.this);
    }
    ....
}
```

**Comment by zoontek** [comment]
@shwanton Your `@drawable/splash_gradient` content is the same as your `bootsplash.xml`. Can you post the correct content? Thanks!

On which Android version + simulator / device do you have the issue?

**Comment by shwanton** [comment]
Fixed the code. I'm seeing the issue on my Emulator (Pixel API 24) & Real Pixel 3.

I'll try to create a repro later today and add screengrab.

**Comment by zoontek** [comment]
Did your app hide the status bar on start? If not, can you try updating your `bootsplash.xml` for

```xml
<?xml version="1.0" encoding="utf-8"?>

<layer-list xmlns:android="http://schemas.android.com/apk/res/android" android:opacity="opaque">
  <item android:drawable="@drawable/splash_gradient" />

  <item>
    <bitmap
      android:src="@mipmap/splash_logo"
      android:gravity="center" />
  </item>
</layer-list>
```

**Comment by zoontek** [comment]
Got it! You used `<item name="android:windowBackground">@drawable/bootsplash</item>`. Replace it with `<item name="android:background">@drawable/bootsplash</item>`, as mentioned in the README.

Don't ignore my previous comment: it fixes an issue for Android < 5 scaling the image on the full screen.

**Comment by shwanton** [comment]
I re-read the docs and you are right, I used the wrong background property!
This did fix the jump on the emulator and my Pixel 3, unfortunately I still see the issue on a Galaxy S7.

I'm also seeing the scaling issue on a Galaxy S4.

When I get time, I'll link a git repo w/ examples.

**Comment by shwanton** [comment]
I just tested this again today and looks like it's working on all devices!
Thank you for the help.

I'm still seeing the scaling issues on a Galaxy S4 but I can open a separate issue for that.

**Comment by zoontek** [comment]
Sometimes it's just the Gradle cache 😄
No problem, open a new issue, we will investigate!

**Comment by vijju1918** [comment]
Im am seeing this issue on redmi note 4 device so i adjusted margintop as -24 and fixed the issue ,but the screens with larger resolution issue still remains.and when i remove margin top -24 (vice versa happens).Please help me out here thanks

**Comment by maderesponsively** [comment]
I added the following to bootsplash.xml and it solved the height shift issue..

`<resources>

    
    <style name="AppTheme" parent="Theme.AppCompat.Light.NoActionBar">
        
        <item name="android:textColor">#000000</item>
        <item name="android:windowFullscreen">true</item> <--- Added this
    </style>

      <style name="BootTheme" parent="AppTheme">
        
        <item name="android:background">@drawable/bootsplash</item>
    </style>

</resources>`

**Comment by iamacoderguy** [comment]
I still got the issue on Xiaomi Redmi

**Comment by iamacoderguy** [comment]
> I still got the issue on Xiaomi Redmi

Nevermind. I've fixed it using the guideline here https://github.com/zoontek/react-native-bootsplash/issues/52#issuecomment-580366701

**Comment by maystrenkoYurii** [comment]
Hi, the best solution I've spent 3 days of my life on https://stackoverflow.com/questions/57461584/centered-image-on-windowbackground-is-different-than-on-an-inflated-layout
![Снимок экрана от 2020-07-06 21-08-53](https://user-images.githubusercontent.com/23016311/86625227-6ad89080-bfcd-11ea-98b1-20580632862b.png)

**Comment by Magudeshwaran** [comment]
> > I still got the issue on Xiaomi Redmi
> 
> Nevermind. I've fixed it using the guideline here [#52 (comment)](https://github.com/zoontek/react-native-bootsplash/issues/52#issuecomment-580366701)

Can you share your styles.xml
